lib/web/websocket/connection.js

```diff
@@ -176,9 +177,19 @@ function establishWebSocketConnection (url, protocols, ws, onEstablish, options)
       //    the WebSocket Connection_.
       const secProtocol = response.headersList.get('Sec-WebSocket-Protocol')
 
-      if (secProtocol !== null && secProtocol !== request.headersList.get('Sec-WebSocket-Protocol')) {
-        failWebsocketConnection(ws, 'Protocol was not set in the opening handshake.')
-        return
+      if (secProtocol !== null) {
+        const requestProtocols = getDecodeSplit('sec-websocket-protocol', request.headersList)
+        const responseProtocols = getDecodeSplit('sec-websocket-protocol', response.headersList)
```

Comment:
server must respond with a single value or with null (equivalent to no header), not with a list